plugins/modules/cloud_provider_account.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: cloud_provider_account
+short_description: Manages an RHSM Cloud Access provider account
+description: >
+  This module will ensure a cloud provider account exists in RHSM and optionally
+  verifies it for use with auto registration.
+options:
+  state:
+    description:
+      - Add or remove a cloud provider account
+    choices: [ present, absent ]
+    default: present
+    type: str
+  refresh_token:
+    description:
+      - Offline refresh token for authentication with Red Hat APIs
+      - Tokens can be generated at https://access.redhat.com/management/api
+    required: true
+    type: str
+  refresh_token_client:
+    description:
+      - Client ID used to generate the refresh token
+      - Typically 'rhsm-api' for tokens generated at access.redhat.com and 'cloud-services' for cloud.redhat.com
+    default: rhsm-api
+    type: str
+  provider:
+    description:
+      - Short Name of the cloud provider
+    choices: [ AWS, MSAZ, GCP ]
```

Comment:
Should we drop `GCP` here given it's not supported yet?

Reply:
"it's still useful in the module since you can create a provider without verifying it"